test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -0,0 +1,352 @@
+    Test that based on admin's input only limited number of resources are
+    released from a running job.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # Set default resources available on the default mom
+        a = {'resources_available.ncpus': 4, 'resources_available.mem': '2gb'}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # Create an express queue
+        b = {'queue_type': 'Execution', 'enabled': 'True',
+             'started': 'True', 'priority': '200'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, b, "expressq")
+
+    def test_do_not_release_mem_sched_susp(self):
+        """
+        During preemption by suspension test that only ncpus are released from
+        the running job and memory is not released.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # Submit a high priority job
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:mem=512mb',
+             ATTR_q: 'expressq'})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S', 'substate': 45}, id=jid1)
+
+        ras_mem = ATTR_rescassn + '.mem'
+        ras_ncpus = ATTR_rescassn + '.ncpus'
+
+        rv = self.server.status(
+            NODE, [ras_ncpus, ras_mem], id=self.mom.shortname)
+        self.assertNotEqual(rv, None)
+
+        self.assertEqual(rv[0][ras_mem], "1048576kb",
+                         msg="pbs should not release memory")
+        self.assertEqual(rv[0][ras_ncpus], "2",
+                         msg="pbs did not release ncpus")
+
+    def test_do_not_release_mem_qsig_susp(self):
+        """
+        If a running job is suspended using qsig, test that only ncpus are
+        released from the running job and memory is not released.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        rc = 0
+        # suspend job
+        try:
+            rc = self.server.sigjob(jobid=jid1, signal="suspend")
+        except PbsSignalError, e:
+            self.assertEqual(rc, 0, e.msg[0])
```

Comment:
Won't this already happen if you don't catch the exception?  rv will not be 0 or the exception wouldn't have been raised.